java-checks/src/main/java/org/sonar/java/checks/RandomFloatToIntCheck.java

```diff
@@ -63,19 +63,26 @@ public List<Tree.Kind> nodesToVisit() {
   @Override
   public void visitNode(Tree tree) {
     TypeCastTree castTree = (TypeCastTree) tree;
-    if(castTree.type().symbolType().is("int")) {
-      castTree.expression().accept(new RandomDoubleVisitor());
+    if (castTree.type().symbolType().is("int")) {
+      castTree.expression().accept(new RandomDoubleVisitor("nextInt()"));
+    } else if (castTree.type().symbolType().is("long")) {
+      castTree.expression().accept(new RandomDoubleVisitor("nextLong()"));
     }
   }
 
   private class RandomDoubleVisitor extends BaseTreeVisitor {
+    private final String methodToCall;
+
+    public RandomDoubleVisitor(String methodToCall){
+      this.methodToCall = methodToCall;
+    }
 
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
       if (MATH_RANDOM_METHOD_MATCHER.matches(tree)) {
-        reportIssue(tree.methodSelect(), "Use \"java.util.Random.nextInt()\" instead.");
```

Comment:
Let's reformat this line to add spaces between `+` and operands to make the concatenation more readable